### PR TITLE
Update plugin authoring guide per style guide

### PR DESCRIPTION
**Pull request by marcysutton**
## Description

I saw https://github.com/gatsbyjs/gatsby/pull/19413 was merged in without explaining some of the concepts, so I followed it up with a PR to add more explanations and links to terminology, including a new Glossary entry for inference (something I answered recently for someone on Twitter after discussing internally). In general, we need to explain jargon and omit words that assume a certain skill/experience level ("just do X", "quickly", "simply", etc.).

## Related Issues

Somewhat related to https://github.com/gatsbyjs/gatsby/issues/16095, but there are more suggestions to work in from @DSchau's prior look at it. (I'm following up on that separately)

### CODE REVIEW

File: docs/docs/creating-a-source-plugin.md [hunk 1]
```diff
@@ -55,30 +55,30 @@ Each node created by the filesystem source plugin includes the
 raw content of the file and its _media type_.
 
 [A **media type**](https://en.wikipedia.org/wiki/Media_type) (also **MIME type**
-and **content type**) are an official way to identify the format of
-files/content that is transmitted on the internet e.g. over HTTP or through
-email. You're probably familiar with many media types such as
+and **content type**) is an official way to identify the format of
+files/content that is transmitted on the internet, e.g. over HTTP or through
+email. You might be familiar with many media types such as
```
Comment (laurieontech) [inline docs/docs/creating-a-source-plugin.md]:
```suggestion
email. You might be familiar with other media types such as
```

File: docs/docs/creating-a-source-plugin.md [hunk 2]
```diff
@@ -92,7 +92,7 @@ not boilerplate.
 
 ## Gotcha: don't forget to return!
 
-After your plugin is finished sourcing nodes, it should either return a promise or use the callback (3rd parameter) to report back to Gatsby when `sourceNodes` is fully executed. If a promise or callback isn't returned, Gatsby will continue on in the build process, before nodes are finished being created. Your nodes might not end up in the generated schema at compilation, or the process will hang while waiting for an indication that it's finished.
+After your plugin is finished sourcing nodes, it should either return a Promise or use the callback (3rd parameter) to report back to Gatsby when `sourceNodes` is fully executed. If a Promise or callback isn't returned, Gatsby will continue on in the build process, before nodes are finished being created. Without the necessary return your nodes might not end up in the generated schema at compilation time, or the process will hang while waiting for an indication that it's finished.
```
Comment (laurieontech) [inline docs/docs/creating-a-source-plugin.md]:
```suggestion
After your plugin is finished sourcing nodes, it should either return a Promise or use the callback (3rd parameter) to report back to Gatsby when `sourceNodes` is fully executed. If a Promise or callback isn't returned, Gatsby will continue on in the build process, before nodes are finished being created. Without the necessary return statement your nodes might not end up in the generated schema at compilation time, or the process will hang while waiting for an indication that it's finished.
```

**Comment by laurieontech** [comment]
Great additional context! Added a couple minor words for clarity, but think this looks good. It's certainly dense, but some content needs to be.

File: docs/docs/creating-a-source-plugin.md [hunk 3]
```diff
@@ -13,16 +13,16 @@ Once a source plugin brings data into Gatsby's system, it can be transformed fur
...
+  [contentDigests](/docs/node-interface/#contentdigest).
 - Links nodes & creates relationships between them.
 - Lets Gatsby know when nodes are finished sourcing so it can move on to processing them.
 
 ## What does the code look like?
 
-A source plugin is a normal NPM package. It has a package.json with optional
-dependencies as well as a `gatsby-node.js` where you implement Gatsby's Node.js
-APIs.
+A source plugin is a regular NPM package. It has a `package.json` file with optional
```
Comment (LekoArts) [inline docs/docs/creating-a-source-plugin.md]: If we want to make things super accurate we'd also need to mention here that a plugin also needs a `main` key in its `package.json` file that points to a `index.js` file that can export components/files or just be empty with a `//noop`. Alternatively this `main` field would point to the mentioned `gatsby-node.js` file.

Comment (m-allanson) [reply docs/docs/creating-a-source-plugin.md]: This is probably ok to leave out from this section? As it's covered in the linked _Files Gatsby Looks for in a Plugin_.

Comment (LekoArts) [reply docs/docs/creating-a-source-plugin.md]: Ah, forgot that it was added there. Scratch that comment then :)

**Comment by marcysutton** [comment]
Thanks for the reviews!

**Pull request merged**